client/components/changed-badge.js

```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import { connect } from 'react-redux';
+
+const ChangedBadge = ({ fields = [], latest = [], granted = [] }) => {
+  const hasChangedFromLatest = fields.some(field => latest.some(change => change.includes(field)));
```

Comment:
I'd add a check that there are actually _any_ changes before iterating over the fields here. That way we don't end up doing a bunch of iterations that we _know_ are not required.

```suggestion
  const hasChangedFromLatest = latest.length && fields.some(field => latest.some(change => change.includes(field)));
```